superset/viz.py

```diff
@@ -1932,9 +1935,18 @@ def process_spatial_data_obj(self, key, df):
         if spatial is None:
             raise ValueError(_('Bad spatial key'))
         if spatial.get('type') == 'latlong':
-            df[key] = list(zip(df[spatial.get('lonCol')], df[spatial.get('latCol')]))
+            df[key] = list(zip(
+                pd.to_numeric(df[spatial.get('lonCol')], errors='coerce'),
+                pd.to_numeric(df[spatial.get('latCol')], errors='coerce'),
+            ))
         elif spatial.get('type') == 'delimited':
-            df[key] = (df[spatial.get('lonlatCol')].str.split(spatial.get('delimiter')))
+
+            def tupleify(s):
+                p = Point(s)
+                return (p.latitude, p.longitude)
+
+            df[key] = df[spatial.get('lonlatCol')].apply(tupleify)
```

Comment:
We should definitely benchmark this with our current data before deploying, since it's using regular expressions to do the job: https://github.com/geopy/geopy/blob/master/geopy/point.py#L310-L339

Reply:
```python
In [2]: from geopy.point import Point

In [3]: Point('234,239')
Out[3]: Point(54.0, -121.0, 0.0)

In [4]: %timeit Point('234,239')
The slowest run took 4.29 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 11.7 µs per loop

In [5]: %timeit (float(v) for v in '234,239'.split(','))
The slowest run took 6.84 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 593 ns per loop
```
So roughly double the time. On 1M points, that means 0.5 second which to me is fine as almost negligible compare to the network time it takes to bring that over. Note that there's probably a numpy way of doing this that would be much faster.

Reply:
Wait, you mean 20x the time. Calling `Point` on 1M points would take ~12 seconds (11.7e-6 s * 1e6).

I looked at the code and I'm not sure how to optimize this with Numpy.

Reply:
Oh snap. 12 seconds is long.